src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -137,13 +137,9 @@ static Lazy<HostDiagnosticAnalyzers> CreateLazyHostDiagnosticAnalyzers(IReadOnly
 
         public SolutionState WithNewWorkspace(Workspace workspace, int workspaceVersion)
         {
-            var services = workspace != _solutionServices.Workspace
-                ? new SolutionServices(workspace)
-                : _solutionServices;
-
             // Note: this will potentially have problems if the workspace services are different, as some services
             // get locked-in by document states and project states when first constructed.
```

Comment:
this comment is outright terrifying (and true).  the workspace services are passed deeply down the solution.  but when we do WithNewWorkspace, we only have a shallow pointer to the new services here.  and all our children still point to the old services =-o